### PR TITLE
Add leader-follower to Solr ECS service

### PR DESCRIPTION
**Pull request by mogul**
@jbrown-xentity and @FuhuXia: Can you confirm that there's no TLS on the connections to the Solr port (8983) on all the instances?

### CODE REVIEW

**Comment by FuhuXia** [comment]
Looking at this [code](https://github.com/GSA/datagov-brokerpak-solr/blob/main/terraform/ecs/provision/lb-follower.tf#L101-L107), within the VPC it is communicating using HTTP, no TLS.

File: ssb/solr.ecs.boundary.md [hunk 1]
```diff
@@ -47,12 +48,15 @@ Rel(aws_cg_alb, cloudgov_router, "proxies requests", "https GET/POST (443)")
...
-Rel(solr_app, solr_instance, "provisions", "https GET/POST (443)")
+Rel(solr_ingress, solr_leader, "proxies requests", "http POST (8983)")
+Rel(solr_ingress, solr_followers, "proxies requests", "http GET (8983)")
+Rel(solr_admin_init_service, solr_leader, "config initialization", "http POST (8983)")
+Rel(solr_app, solr_leader, "provisions", "https GET/POST (443)")
 Rel(solr_app, solr_ingress, "provisions", "https GET/POST (443)")
 Rel(solr_app, solr_admin_init_service, "provisions", "https GET/POST (443)")
 Rel(solr_app, solr_efs, "provisions", "https GET/POST (443)")
-Rel(solr_instance, solr_efs, "mounts", "nfs")
+Rel(solr_leader, solr_followers, "distributes updates", "http POST (8983)")
```
Comment (FuhuXia) [inline ssb/solr.ecs.boundary.md]:
```suggestion
Rel(solr_leader, solr_followers, "distributes updates", "https POST (443)")
```

Comment (FuhuXia) [reply ssb/solr.ecs.boundary.md]: HTTPS/443 is used for replication. 
https://github.com/GSA/datagov-brokerpak-solr/blob/main/terraform/ecs/provision/follower.tf#L29

**Comment by mogul** [comment]
> Looking at this [code](https://github.com/GSA/datagov-brokerpak-solr/blob/main/terraform/ecs/provision/lb-follower.tf#L101-L107), within the VPC it is communicating using HTTP, no TLS.

This is really surprising! If the leader talks to the followers over TLS on 443, is there any reason that the ALB can't do the same? ("TLS everywhere" is a big compliance sticking point.)

**Comment by FuhuXia** [comment]
It is multiple followers talking to leader ALB on port 443. Leader ALB still talks to the single leader on port 8983. So maybe we should delete this line, or change it to
`Rel(solr_followers, solr_leader, "fetch updates", "https POST (443)")`

https://github.com/GSA/catalog.data.gov/blob/main/ckan/setup/solr/solrconfig_follower.xml#L301
https://github.com/GSA/datagov-brokerpak-solr/blob/main/terraform/ecs/provision/follower.tf#L29

**Comment by mogul** [comment]
Made updates; should be better, let me know what you think!

**Review by FuhuXia — APPROVED** [review]

**Pull request merged**